app/controllers/submissions_controller.rb

```diff
@@ -509,39 +507,42 @@ def set_testrun
     }
   end
 
-  def transmit_lti_score
-    responses = send_scores(@submission)
-    failed_users = []
-    too_late = false
-    status = []
-    score_sent = nil
-
-    responses.each do |response|
-      if Lti::ERROR_STATUS.include? response[:status]
-        failed_users << response[:user]
-      elsif response[:score_sent] != @submission.normalized_score # the score was sent successfully, but received too late
-        too_late = true
-        score_sent = response[:score_sent]
-      end
+  def check_scoring_too_late(submit_info, client_socket)
+    if submit_info[:deadline] == :within_grace_period || submit_info[:deadline] == :after_late_deadline
+      client_socket&.send_data({
+        cmd: :status,
+        status: :scoring_too_late,
+        score_sent: submit_info[:score][:sent],
+      }.to_json)
     end
+  end
 
-    if failed_users.size == responses.size # all submissions failed
-      status.push(:scoring_failure)
-      {status:}
-    elsif too_late & failed_users.size.positive? # submissions are too late and at least one submission failed
-      status.push(:too_late_and_not_for_all_users_submitted)
-      failed_users = failed_users.join(', ')
-      {status:, score_sent:, failed_users:}
-    elsif failed_users.size.positive? # at least one submission failed
-      status.push(:not_for_all_users_submitted)
-      failed_users = failed_users.join(', ')
-      {status:, failed_users:}
-    elsif too_late # submissions are too late
-      status.push(:scoring_too_late)
-      {status:, score_sent:}
-    elsif @submission.score == @submission.exercise.maximum_score
-      status.push(:full_score_reached)
-      {status:}
+  def transmit_lti_score(client_socket)
+    submit_info = send_scores(@submission)
+
+    if (submit_info[:users][:all].count == submit_info[:users][:success].count) & (submit_info[:score][:original] == BigDecimal('1.0'))
```

Comment:
Here, it might make sense to use a case-when construct. This could allow to clean up the code and will probably also avoid redoing the same calculations over and over again (in the current implementation, the `submit_info[:users][:all].count` is performed multiple times, even though the value is unlikely to change between the cases.

Reply:
The conditions changed so that it is now easier to use an if-else construct.